src/internal.c

```diff
@@ -6414,6 +6427,14 @@ void SSL_ResourceFree(WOLFSSL* ssl)
      * example with the RNG, it isn't used beyond the handshake except when
      * using stream ciphers where it is retained. */
 
+#ifdef HAVE_EX_DATA_CLEANUP_HOOKS
+    {
+        int idx;
+        for (idx = 0; idx < MAX_EX_DATA; ++idx)
```

Comment:
This code pattern is duplicated. Perhaps a `wolfSSL_CRYPTO_set_ex_data_with_cleanup` version that free's all indexes or actually track the max index?

Reply:
Good idea!  Done.